### PR TITLE
Convert even more ZStream combinators to process

### PR DESCRIPTION
**Pull request by iravid**
@vasilmkd

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -557,8 +558,8 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
    * Appends another stream to this stream. The concatenated stream will first emit the
    * elements of this stream, and then emit the elements of the `other` stream.
    */
-  final def concat[R1 <: R, E1 >: E, A1 >: A](other: ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
-    Stream(self, other).flatMap(identity)
+  final def concat[R1 <: R, E1 >: E, A1 >: A](other: => ZStream[R1, E1, A1]): ZStream[R1, E1, A1] =
+    ZStream(UIO.succeed(self), UIO(other)).flatMap(ZStream.unwrap)
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: @jdegoes is there another combinator constellation we could use to make this nicer? e.g. a lazy constructor like `construct(head: A, rest: => List[A])`

Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Awesome trick.

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 2]
```diff
@@ -909,17 +909,7 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
...
-        ZManaged.succeed { (s, cont, f) =>
-          def loop(s: S): ZManaged[R1, E1, S] =
-            self.fold[R1, E1, A, S].flatMap { fold =>
-              fold(s, cont, f).flatMap(s => if (cont(s)) loop(s) else ZManaged.succeed(s))
-            }
-
-          loop(s)
-        }
-    }
+    self ++ forever
```
Comment (vasilmkd) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: And it's logical conclusion.

**Comment by vasilmkd** [comment]
Looks good to me.

**Pull request merged**